airflow/providers/amazon/aws/sensors/sqs.py

```diff
@@ -102,3 +152,39 @@ def get_hook(self) -> SQSHook:
 
         self.hook = SQSHook(aws_conn_id=self.aws_conn_id)
         return self.hook
+
+    def filter_messages(self, messages):
+        if self.message_filtering == 'literal':
+            return self.filter_messages_literal(messages)
+        if self.message_filtering == 'jsonpath':
+            return self.filter_messages_jsonpath(messages)
+        else:
+            raise NotImplementedError('Override this method to define custom filters')
+
+    def filter_messages_literal(self, messages):
+        filtered_messages = []
+        if self.message_filtering_match_values is None:
+            raise Exception('message_filtering_match_values must be specified for literal matching')
```

Comment:
This should be checked in `__init__`, and shoul raise a `TypeError` instead.

Reply:
Fixed, thanks for the feedback